File: January Challenge/Python/Evaluate reverse polish notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        Q = []
        st = []
        op = {"+","-","*","/"}

        for i in tokens:
            if i not in op:
                Q.append(int(i))
            else:
                st.append(i)

            while st and len(Q) > 1:
                operator = st.pop()
                if operator == "+":
                    res = int(Q.pop(-2)) + int(Q.pop(-1))

                elif operator == "-":
                    res = int(Q.pop(-2)) - int(Q.pop(-1))

                elif operator == "*":
                    res = int(Q.pop(-2)) * int(Q.pop(-1))
                
                else:
                    res = int((Q.pop(-2)) / (Q.pop(-1)))

                Q.append(res) 
        
        return Q.pop()

File: January Challenge/Java/Evaluate reverse polish notation.java
import java.util.Stack;
class Solution {
    public int evalRPN(String[] tokens) {
        ArrayList<Integer> Q = new ArrayList<>();
        Stack<String> st = new Stack();
        HashSet<String> op = new HashSet<>();
        op.add("+");
        op.add("-");
        op.add("*");
        op.add("/");

        for(String i: tokens){
            if(!op.contains(i)) Q.add(Integer.parseInt(i));
            
            else st.push(i);
            

            while(!st.empty() && Q.size() > 1){
                String operator = st.pop();
                int res = 0;
                if (operator.equals("+")) res = Q.get(Q.size() - 2) + Q.get(Q.size() - 1);
                

                else if (operator.equals("-")) res = Q.get(Q.size() - 2) - Q.get(Q.size() - 1);
                
                
                else if (operator.equals("*")) res = Q.get(Q.size() - 2) * Q.get(Q.size() - 1);
                
                else res = Q.get(Q.size() - 2) / Q.get(Q.size() - 1);
                
                
                Q.remove(Q.size() - 1);
                Q.remove(Q.size() - 1);
                Q.add(res);
            }
        }
        return Q.get(0);
    }
}
